### ISSUE TITLE
"%" in `battery#component()` is escaped and that cause "%%" when the component is used in `%{}` expression

### ISSUE THREAD

**Issue by Freed-Wu** [lambdalisue/battery.vim#22]
```vim
#!/usr/bin/env -S vi -u
" $ uname -r
" 6.0.2-arch1-1
" $ has
" ✓ vi 0.8.0
" $ cat test.vim
set runtimepath=$VIMRUNTIME
set runtimepath+=~/.local/share/nvim/repos/github.com/lambdalisue/battery.vim
set statusline=%{battery#component()}
" $ chmod +x test.vim
" $ ./test.vim
```

![screen-2022-11-05-20-10-54](https://user-images.githubusercontent.com/32936898/200119332-3fd432ea-7d02-43f7-a084-243702e92035.jpg)

Expected: single %, not double.

**Comment by lambdalisue** [comment]
Hum... Actually, it's intentional.

```
			   *'statusline'* *'stl'* *E540* *E542*
'statusline' 'stl'	string	(default empty)
			global or local to window |global-local|
	When non-empty, this option determines the content of the status line.
	Also see |status-line|.

	The option consists of printf style '%' items interspersed with
	normal text.  Each status line item is of the form:
	  %-0{minwid}.{maxwid}{item}
	All fields except the {item} are optional.  A single percent sign can
	be given as "%%".
```

Not sure but maybe using `%!` change the behavior. I'll investigate.

**Comment by lambdalisue** [comment]
Prob fixed with #24
